src/MatrixTypes_Native.f90

```diff
@@ -1369,6 +1370,7 @@ SUBROUTINE set_BandedMatrixType(matrix,i,j,setval)
   IF((j <= matrix%m) .AND. (i <= matrix%n) .AND. (i >= 1) .AND. (j >= 1)) THEN
     IF(.NOT. matrix%isAssembled) THEN
       ! If it is not assembled, add to tmp variables
+      REQUIRE(matrix%counter < matrix%nnz)
```

Comment:
`<` or `<=` ?

Reply:
It needs to be < because counter is incremented before the set call. Otherwise counter would equal the number of entries plus one, which isn't a disaster but I chose to set it up this way.

Reply:
Perhaps just make a note of that somewhere if you haven't already.